pkg/client/client_test.go

```diff
@@ -0,0 +1,157 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	api "github.com/go-itools-internship/go-secret/pkg/http"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestClient_SetByKey(t *testing.T) {
+	cipherKey := "c-key"
+	getter := "key"
+	method := "cloud"
+	value := "test-value"
+	t.Run("success", func(t *testing.T) {
+		ctx := context.Background()
+		s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			require.EqualValues(t, "/", r.URL.Path)
+			require.EqualValues(t, http.MethodPost, r.Method)
+			require.EqualValues(t, r.Header.Get(api.ParamCipherKey), cipherKey)
```

Comment:
Assume that expected value should be `cipherKey` and actual `r.Header.Get(api.ParamCipherKey)`

Look thought all tests, find and fix the same issues

```suggestion
			require.EqualValues(t, cipherKey, r.Header.Get(api.ParamCipherKey))
```